jina/drivers/cache.py

```diff
@@ -21,23 +21,20 @@ class BaseCacheDriver(BaseIndexDriver):
     def __init__(self, with_serialization: bool = False, *args, **kwargs):
         self.with_serialization = with_serialization
         super().__init__(*args, **kwargs)
-        self.field = None
 
     def _apply_all(self, docs: 'DocumentSet', *args, **kwargs) -> None:
-        self.field = self.exec.field
-
         if self._method_name == 'update':
-            self.exec_fn([d.id for d in docs], [d.id if self.field == ID_KEY else d.content_hash for d in docs])
+            values = [''.join(d.get_attrs(*self.exec.fields).values()) for d in docs]
```

Comment:
If from here u can get a List of Tuples, I think the Tuple can be hashed? Could this may be a better solution?

Reply:
What do you mean?